### PR TITLE
Update dependencies for coverage package

### PR DESCRIPTION
**Pull request by Prusdrum**
## Description

This PR upgrades dependencies. The most important change is switching to ESM because the main package - meow - [requires it](https://github.com/sindresorhus/meow/releases/tag/v10.0.0). 

It wouldn't be a breaking change if not for the Node.js update to v16 - [also required by meow](https://github.com/sindresorhus/meow/releases/tag/v12.0.0). Hence the major version bump.

## Related issues & PRs

related: https://github.com/simpleclub/firebase-rules-helper/pull/23

[keywords]: https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

## Checklist

- [x] I have made myself familiar with the
  [contribution guide](https://github.com/simpleclub/firebase-rules-helper/blob/master/CONTRIBUTING.md).
- [x] I added a PR description.
- [x] I linked all related issues and PRs I could find (no links if there are none).
- [x] I've bumped all versions and created entries in `CHANGELOG.md` for all affected packages.
- [x] If there is new functionality in code, I added tests covering all my additions.
- [x] All required checks pass.

### CODE REVIEW

**Comment by Prusdrum** [comment]
@IchordeDionysos, the "Lints & Tests / firebase-rules-coverage lints on Node 12" job fails because the expected version is 16. Because of `meow` it won't be able to run on neither node 12 nor 14. Should we disable these jobs?

**Comment by IchordeDionysos** [comment]
Yes sure

**Comment by Prusdrum** [comment]
After giving it a second thought, I can see a challenge here, as the workflows are defined to be shared with the other package. I will need to look deeper into how I could achieve it, so that we run the node12 checks for the firebase-rules-generate but don't do it for firebase-rules-coverage

File: .github/workflows/checks-generator.yaml
Comment (alexregier) [inline .github/workflows/checks-generator.yaml:23]: should we remove 12 then?

File: firebase-rules-coverage/src/cli.ts
Comment (alexregier) [inline firebase-rules-coverage/src/cli.ts:5]:
```suggestion
import {generateLcovFile} from './convert';
```

File: firebase-rules-coverage/src/convert.ts
Comment (alexregier) [inline firebase-rules-coverage/src/convert.ts:7]:
```suggestion
} from './interfaces';
```

File: firebase-rules-coverage/src/index.ts
Comment (alexregier) [inline firebase-rules-coverage/src/index.ts:1]:
```suggestion
export {firebaseCoverageToLcov, generateLcovFile} from './convert';
```

File: firebase-rules-coverage/src/interfaces/index.ts
Comment (alexregier) [inline firebase-rules-coverage/src/interfaces/index.ts:2]:
```suggestion
export * from './coverage_results';
export * from './firebase_coverage';
```

File: firebase-rules-coverage/test/cli.test.ts
Comment (alexregier) [inline firebase-rules-coverage/test/cli.test.ts:4]:
```suggestion
type TestedModule = typeof import('../src/convert');
```

File: firebase-rules-coverage/test/cli.test.ts
Comment (alexregier) [inline firebase-rules-coverage/test/cli.test.ts:14]:
```suggestion
    jest.unstable_mockModule('../src/convert', () => ({
```

File: firebase-rules-coverage/test/cli.test.ts
Comment (alexregier) [inline firebase-rules-coverage/test/cli.test.ts:17]:
```suggestion
    const module = await import('../src/convert');
```

File: firebase-rules-coverage/test/cli.test.ts
Comment (alexregier) [inline firebase-rules-coverage/test/cli.test.ts:41]:
```suggestion
    await import('../src/cli');
```

File: firebase-rules-coverage/test/cli.test.ts
Comment (alexregier) [inline firebase-rules-coverage/test/cli.test.ts:66]:
```suggestion
    await import('../src/cli');
```

File: firebase-rules-coverage/test/convert.test.ts
Comment (alexregier) [inline firebase-rules-coverage/test/convert.test.ts:3]:
```suggestion
import {firebaseCoverageToLcov, generateLcovFile} from '../src';
```

**Review by alexregier — APPROVED** [review]

File: .github/workflows/checks-generator.yaml
Comment (Prusdrum) [reply .github/workflows/checks-generator.yaml:23]: This is the coverage package and this file is about generator. Because they are two packages, I will treat them in separate PRs and in the following package I will remove unsupported node versions. As you can see https://github.com/simpleclub/firebase-rules-helper/pull/22/files#diff-5c148a23087f21b63c00dbf193fe76a01cdfc09e6d523ae74e6817f387a2ddb5R25-R26 for the coverage package I've already done it.

**Comment by Prusdrum** [comment]
@IchordeDionysos, could you approve if everything looks ok to you? Your approval is required as you're the code owner.

File: firebase-rules-coverage/src/convert.ts
Comment (alexregier) [reply firebase-rules-coverage/src/convert.ts:7]: okay

File: firebase-rules-coverage/tsconfig.json
Comment (IchordeDionysos) [inline firebase-rules-coverage/tsconfig.json:8]: Or like this?
```suggestion
    "moduleResolution": "node16"
```

**Pull request merged**
